protected/humhub/modules/content/permissions/AbstractContentPermission.php

```diff
@@ -4,9 +4,27 @@
 
 use humhub\components\permission\BasePermission;
 use humhub\modules\content\components\ContentActiveRecord;
+use humhub\modules\content\components\ContentContainerActiveRecord;
+use humhub\modules\content\models\Content;
 use humhub\modules\user\models\User;
 
+/**
+ * @property-read Content $content
+ * @property-read ContentContainerActiveRecord $container
+ */
 abstract class AbstractContentPermission extends BasePermission
 {
-    abstract public function verify(ContentActiveRecord $content, ?User $user): bool;
+    public ContentActiveRecord $model;
+
+    abstract public function verify(?User $user = null): bool;
+
+    public function getContent(): ?Content
+    {
+        return $this->model instanceof ContentActiveRecord ? $this->model->content : null;
+    }
+
+    public function getContainer(): ?ContentContainerActiveRecord
```

Comment:
Can this new methods be `protected`?

I'm not a big fan of such "magic" properties either. 

If it is not possible when the method is protected, then I would omit the properties and use only the method directly. 

Otherwise, I like the methods/properties as they make the code much cleaner.